### PR TITLE
Move code to interact with user out of sypgp 

### LINKED ISSUE #3915: move the code for interactive interactions with users out of sypgp
## Version of Singularity:

master.

### Expected behavior

@mem correctly pointed out that the current `sypgp` code is mixing CLI code, with actual code to handle keys. A first good step would be to move the code specific to asking questions out of `sypgp` to a utility package.

### Actual behavior

Everything is in `sypgp`.

### Steps to reproduce behavior

N/A.

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

- Move code to interact with user out of sypgp to a separate utility package. This is the first step to separate the core code of sypgp from the front-end code.
- Improve test coverage.

Next step will be to refactor the code to separate actual UI code from core GPG code.

**This fixes or addresses the following GitHub issues:**

- Fixes #3915 
- Addresses #2848

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: pkg/sypgp/sypgp.go [hunk 1]
```diff
@@ -40,10 +38,7 @@ const helpAuth = `Access token is expired or missing. To update or obtain a toke
 const helpPush = `  4) Push key using "singularity key push %[1]X"
 `
 
-var errPassphraseMismatch = errors.New("passphrases do not match")
-var errTooManyRetries = errors.New("too many retries while getting a passphrase")
 var errNotEncrypted = errors.New("key is not encrypted")
```
Comment (jscook2345) [inline pkg/sypgp/sypgp.go:44]: should this be moved with the other errors?

**Comment by jscook2345** [comment]
Looks pretty good to me otherwise!

**Review by ikaneshiro — APPROVED** [review]
lgtm

File: pkg/sypgp/sypgp.go (hunk 1 above)
Comment (ikaneshiro) [reply pkg/sypgp/sypgp.go:44]: This error is still used in this package within `DecryptKey()` and `RecryptKey()` so it needs to stay

File: internal/pkg/util/interactive/interactive.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
```
Comment (WestleyK) [inline internal/pkg/util/interactive/interactive.go]: This looks like a new package, so the copyright should be `2019`

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
```

File: internal/pkg/util/interactive/interactive.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+		return "y", nil
+
+	case "n", "no":
+		return "n", nil
+
+	case "":
+		return defaultAnswer, nil
+
+	default:
+		return "", fmt.Errorf("invalid answer %q", ans)
```
Comment (WestleyK) [inline internal/pkg/util/interactive/interactive.go]:
```suggestion
		return "", fmt.Errorf("invalid answer: %q", ans)
```

File: internal/pkg/util/interactive/interactive.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+// AskNumberInRange prompts the user expecting an answer that is a number
+// between start and end.
+func AskNumberInRange(start, end int, format string, a ...interface{}) (int, error) {
+	ans, err := AskQuestion(format, a...)
+	if err != nil {
+		return 0, err
+	}
+	fmt.Println("Answer:", ans)
```
Comment (WestleyK) [inline internal/pkg/util/interactive/interactive.go]: Leftover debugging?

File: internal/pkg/util/interactive/interactive_test.go [hunk 5]
```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
```
Comment (WestleyK) [inline internal/pkg/util/interactive/interactive_test.go]: Same here about the copyright

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
```

File: pkg/sypgp/sypgp.go [hunk 6]
```diff
@@ -865,10 +696,12 @@ func RecryptKey(k *openpgp.Entity, passphrase []byte) error {
 	}
 
 	if err := k.PrivateKey.Decrypt(passphrase); err != nil {
+		fmt.Printf("failed to decrypt key with %s", string(passphrase))
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go]: Why was this added? Can this instead be returned in the error message?

File: pkg/sypgp/sypgp.go [hunk 7]
```diff
@@ -865,10 +696,12 @@ func RecryptKey(k *openpgp.Entity, passphrase []byte) error {
...
 		return err
 	}
 
 	if err := k.PrivateKey.Encrypt(passphrase); err != nil {
+		fmt.Println("failed to encrypt key")
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go]: Same here

**Review by WestleyK — COMMENTED** [review]
Looks good, just a couple comments. Thanks :+1:

**Review by WestleyK — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**